tests/tools/onert_train/src/nnfw_util.h

```diff
@@ -31,11 +32,17 @@
     }                                \
   } while (0)
 
+template <typename E> constexpr auto to_underlying(E e) noexcept
+{
+  return static_cast<std::underlying_type_t<E>>(e);
+}
+
```

Comment:
(note) to find the underlying value(usually integer) for enum
(I referred this function from modern effective c++ item 10)  
 
---------------------------------------

here are some c++ things I checked : 

* about `underlying_type_t` : 
   https://en.cppreference.com/w/cpp/types/underlying_type#:~:text=T%20%3E%0Ausing-,underlying_type_t,-%3D%20typename%20underlying_type 

* `constexpr` implies inline, that's why it defines in header file 
   https://en.cppreference.com/w/cpp/language/constexpr#:~:text=declaration%20implies%20inline

Reply:
I have a plan to use this function here  - https://github.com/Samsung/ONE/pull/12545/files#r1470624916

Reply:
How about moving this function into `runtime/libs/misc/include/misc`? Then we can also use this function in `onert`.

Reply:
+1 for @ragmani.

1. I prefer to create `to_underlying.h` under `runtime/libs/misc/include/misc` just like `polymorphic_downcast.h`.
2. [SO](https://stackoverflow.com/questions/8357240/how-to-automatically-convert-strongly-typed-enum-into-int) says [`std::to_underlying`](https://en.cppreference.com/w/cpp/utility/to_underlying) is already a part of c++ since c++23.

Reply:
Thanks for the detailed guide. :+1: 
Updated.